### PR TITLE
Update framework.rst

### PR DESCRIPTION
**Pull request by AnthonyLim23**
Cleaned up the documentation.

Description of work.

Check the documentation renders correctly and spag. 

Have altered the release notes. 
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by NickDraper** [comment]
Build cancelled as this is rst only - ignore the failures

**Comment by NickDraper** [comment]
build cancelled to save resources

File: docs/source/release/v3.9.0/framework.rst [hunk 1]
```diff
@@ -108,8 +102,8 @@ Improved
 Python
 ------
 
-- The function `IMDDimension.getName()` has been deprecated. Use the propery `IMDDimension.name` instead.
-- The duplicate function `Workspace.getName()` has been deprecated. Use `Workspace.name()` instead.
+- The function `IMDDimension.getName()` has been deprecated in favour of `IMDDimension.name'.
+- The duplicate function `Workspace.getName()` has been deprecated in favour of `Workspace.name()'.
 
 Python Algorithms
```
Comment (NickDraper) [inline docs/source/release/v3.9.0/framework.rst]: Can you get rid of the python algorithm and bug fixes sections and merge them into the new and improved algorithms sections

**Comment by NickDraper** [comment]
@AnthonyLim23 I think you missed the point of my last comment a bit, I also want the Python Algorithms section removed, and merged into the new and updated sections of the "Algorithms" section earlier in the page.

**Comment by NickDraper** [comment]
You conflict now is almost certainly another line has been added that will need fitting into the improved doc.

**Comment by AnthonyLim23** [comment]
Oh ok. Sorry I didn't realise that was what you meant.

**Comment by NickDraper** [comment]
Ready to go :shipit: and merge

**Pull request merged**